### PR TITLE
rbac: expose TLS properties in conditions

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Description: Achieve parity with RBAC matchers in CEL. Refactor to use read-only views of the containers, to avoid copies and managing data ownership.

Risk Level: low
Testing: unit
Docs Changes: updated attribute list
Release Notes: added TLS properties to RBAC conditions

### CODE REVIEW

File: docs/root/intro/arch_overview/security/rbac_filter.rst [hunk 1]
```diff
@@ -89,8 +89,21 @@ The following attributes are exposed to the language runtime:
...
+   connection.dns_san_local_certificate, string, The first DNS entry in the SAN field of the local certificate in the downstream TLS connection
+   connection.dns_san_peer_certificate, string, The first DNS entry in the SAN field of the peer certificate in the downstream TLS connection
+   connection.uri_san_local_certificate, string, The first URI entry in the SAN field of the local certificate in the downstream TLS connection
+   connection.uri_san_peer_certificate, string, The first URI entry in the SAN field of the peer certificate in the downstream TLS connection
    upstream.address, string, Upstream connection remote address
    upstream.port, int, Upstream connection remote port
+   upstream.tls_version, string, TLS version of the upstream TLS connection
+   upstream.subject_local_certificate, string, The subject field of the local certificate in the upstream TLS connection
+   upstream.subject_peer_certificate, string, The subject field of the peer certificate in the upstream TLS connection
+   upstream.dns_san_local_certificate, string, The first DNS entry in the SAN field of the local certificate in the upstream TLS connection
```
Comment (bianpengyuan) [inline docs/root/intro/arch_overview/security/rbac_filter.rst]: Shall we return a list instead of just the first one?

Comment (kyessenov) [reply docs/root/intro/arch_overview/security/rbac_filter.rst]: RBAC, ext authz, and istio use only the first one. Lists would require allocating an extra wrapper unfortunately, so let's not do that until someone asks.

**Comment by kyessenov** [comment]
xref https://github.com/istio/istio/issues/18139

**Pull request merged**